File: algorithms/medium/minimum_operations_to_make_binary_array_elements_equal_to_one_1.go
package main

import "fmt"

func minOperations(nums []int) int {
    N := len(nums)
    res := 0
    for i := 0; i < N - 2; i++ {
        if nums[i] == 0 {
            nums[i] = 1
            if nums[i+1] == 0 {
                nums[i+1] = 1
            } else {
                nums[i+1] = 0
            }
            if nums[i+2] == 0 {
                nums[i+2] = 1
            } else {
                nums[i+2] = 0
            }
            res++
        }
    }
    if nums[N-1] == 0 || nums[N-2] == 0 {
        return -1
    }
    return res
}

func main() {
	data := [][]int{
                 []int{0,1,1,1,0,0},
                 []int{0,1,1,1},
                 []int{0,0,0,0,0,0,0,0,0},
                 []int{1,1,1,1,1,1,1,1,1},
                 []int{1,0,0,0,1,0,0,0,1},
	        }
	for _, nums := range data {
		fmt.Printf("nums = %v\n", nums)
		r := minOperations(nums)
		fmt.Printf("r = %d\n", r)
		fmt.Println("==========================")
	}
}

File: algorithms/medium/minimum_operations_to_make_binary_array_elements_equal_to_one_1.py
from typing import List

class Solution:
    def minOperations(self, nums: List[int]) -> int:
        N = len(nums)
        flips = [0 for _ in range(N)]
        i = 0
        res = 0
        while i < N:
            while i < N and (nums[i] + flips[i]) % 2 == 1:
                i += 1
            if N - i < 3:
                break
            for j in range(3):
                flips[i+j] += 1
            res += 1
        return -1 if i < N else res

# Main section
for nums in [
               [0,1,1,1,0,0],
               [0,1,1,1],
               [0,0,0,0,0,0,0,0,0],
               [1,1,1,1,1,1,1,1,1],
               [1,0,0,0,1,0,0,0,1],
            ]:
    print(f'nums = {nums}')
    sol = Solution()
    r = sol.minOperations(nums)
    print(f'r  = {r}')
    print('================')

